public_html/wp-content/mu-plugins/blocks/assets/src/sessions/edit.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, pickBy, split } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+const apiFetch = wp.apiFetch;
+const { withSelect } = wp.data;
+const { Component, Fragment } = wp.element;
+const { addQueryArgs } = wp.url;
+
+/**
+ * Internal dependencies
+ */
+import SessionsBlockControls from './block-controls';
+import SessionsInspectorControls from './inspector-controls';
+import GridToolbar from '../shared/grid-layout/toolbar';
+
+const blockData = window.WordCampBlocks.sessions || {};
+
+const SESSIONS_ICON = 'list-view';
+const MAX_POSTS = 100;
+
+const ALL_POSTS_QUERY = {
+	orderby  : 'title',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+	_embed   : true,
+};
+
+const ALL_TERMS_QUERY = {
+	orderby  : 'name',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+};
+
+class SessionsEdit extends Component {
+	constructor( props ) {
+		super();
+
+		this.state = {
+			allSessionPosts      : null,
+			allSessionTracks     : null,
+			allSessionCategories : null,
+		};
+	}
+
+	componentWillMount() {
+		this.isStillMounted = true;
+
+		const allSessionPosts = apiFetch( {
+			path: addQueryArgs( `/wp/v2/sessions`, ALL_POSTS_QUERY ),
+		} );
+		const allSessionTracks = apiFetch( {
+			path: addQueryArgs( `/wp/v2/session_track`, ALL_TERMS_QUERY ),
+		} );
+		const allSessionCategories = apiFetch( {
+			path: addQueryArgs( `/wp/v2/session_category`, ALL_TERMS_QUERY ),
+		} );
+
+		if ( this.isStillMounted ) {
+			this.setState( {
+				allSessionPosts      : allSessionPosts, // Promise
+				allSessionTracks     : allSessionTracks, // Promise
+				allSessionCategories : allSessionCategories, // Promise
+			} );
+		}
+	}
+
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
+	render() {
+		const { mode } = this.props.attributes;
+
+		return (
+			<Fragment>
+				<SessionsBlockControls
+					icon={ SESSIONS_ICON }
+					{ ...this.props }
+					{ ...this.state }
+				/>
+				{ mode &&
```

Comment:
This feels like of unintuitive to me, like from the context I expect `mode` to be a boolean, but then it's name doesn't describe what it is, so I'm left confused.

I had to look through other parts of the code to realize it's a string that has an empty value until a mode like "all speakers" or "group of speakers" is selected.

I wonder if the code would be more descriptive/self-documenting/obvious if we gave `mode` an initial value of `placeholder`, and then changed things like this to read `'placeholder' !== mode` ?

@coreymckrill , @vedanshujain

Reply:
That would be fine with me.